### PR TITLE
Updated KeyVault data-plane library to new structure

### PR DESCRIPTION
**Pull request by chidozieononiwu**
Moved KeyVault data-plane library into sdk folder.
Updated Build properties for Data-Plane projects

### CODE REVIEW

**Comment by chidozieononiwu** [comment]
@Azure/azure-sdk-eng

File: Directory.Build.props [hunk 1]
```diff
@@ -6,6 +6,7 @@
     <IsClientLibrary Condition="'$(IsClientLibrary)' == '' and $(MSBuildProjectName.StartsWith('Azure.'))">true</IsClientLibrary>
     <IsDataPlaneProject Condition="'$(IsDataPlaneProject)' == '' and '$(IsClientLibrary)' == 'true'">true</IsDataPlaneProject>
     <IsDataPlaneProject Condition="'$(IsDataPlaneProject)' == '' and $(MSBuildProjectDirectory.Contains('data-plane'))">true</IsDataPlaneProject>
+    <IsDataPlaneProject Condition="'$(IsDataPlaneProject)' == '' and $([System.Text.RegularExpressions.Regex]::IsMatch('$(MSBuildProjectDirectory)', '.*(\\sdk\\.*?\\Microsoft.Azure..*?\\)'))">true</IsDataPlaneProject>
```
Comment (weshaggard) [inline Directory.Build.props]: No need to use the regex as that isn't as efficient just make this condition `$(MSBuildProjectName.StartsWith('Microsoft.Azure.'))`

Comment (weshaggard) [reply Directory.Build.props]: We will likely also need to add some exclusions for management libraries.

Comment (chidozieononiwu) [reply Directory.Build.props]: Ok, that why I used the regex instead. I will try to get it working that way.

File: sdk/AzSdk.reference.props [hunk 2]
```diff
@@ -0,0 +1,36 @@
+<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+	<Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.props'))" />
```
Comment (weshaggard) [inline sdk/AzSdk.reference.props]: Why are these reference.props files needed? Should the one under eng be getting picked up?

File: sdk/keyvault/Microsoft.Azure.KeyVault.Extensions/tests/Microsoft.Azure.KeyVault.Extensions.Tests.csproj [hunk 3]
```diff
@@ -12,11 +12,11 @@
...
-    <ProjectReference Include="..\Microsoft.Azure.KeyVault\Microsoft.Azure.KeyVault.csproj" />
-    <ProjectReference Include="..\Microsoft.Azure.KeyVault.Cryptography\Microsoft.Azure.KeyVault.Cryptography.csproj" />
-    <ProjectReference Include="..\Microsoft.Azure.KeyVault.Extensions\Microsoft.Azure.KeyVault.Extensions.csproj" />
-    <ProjectReference Include="..\Microsoft.Azure.KeyVault.Core\Microsoft.Azure.KeyVault.Core.csproj" />
-    <ProjectReference Include="..\Microsoft.Azure.KeyVault.TestFramework\Microsoft.Azure.KeyVault.TestFramework.csproj" />
+    <ProjectReference Include="..\..\Microsoft.Azure.KeyVault\src\Microsoft.Azure.KeyVault.csproj" />
+    <ProjectReference Include="..\..\Microsoft.Azure.KeyVault.Cryptography\src\Microsoft.Azure.KeyVault.Cryptography.csproj" />
+    <ProjectReference Include="..\..\Microsoft.Azure.KeyVault.Extensions\src\Microsoft.Azure.KeyVault.Extensions.csproj" />
+    <ProjectReference Include="..\..\Microsoft.Azure.KeyVault.Core\src\Microsoft.Azure.KeyVault.Core.csproj" />
+    <ProjectReference Include="..\..\Microsoft.Azure.KeyVault.TestFramework\src\Microsoft.Azure.KeyVault.TestFramework.csproj" />
```
Comment (weshaggard) [inline sdk/keyvault/Microsoft.Azure.KeyVault.Extensions/tests/Microsoft.Azure.KeyVault.Extensions.Tests.csproj]: We shouldn't have the TestFramework project under the root of keyvault. We should move it under one of the other tests folders. I would suggest putting it in a sub-folder under Microsoft.Azure.KeyVault\tests\TestFramework. Note if you do that you will need to exclude that directory from being automatically added to the Microsoft.Azure.KeyVault.Tests project.

File: sdk/keyvault/Microsoft.Azure.KeyVault.WebKey/tests/Microsoft.Azure.KeyVault.WebKey.Tests.csproj [hunk 4]
```diff
@@ -6,7 +6,7 @@
   </PropertyGroup>
   <ItemGroup>
     <Reference Include="Microsoft.CSharp" />
-    <ProjectReference Include="..\Microsoft.Azure.KeyVault.WebKey\Microsoft.Azure.KeyVault.WebKey.csproj" />
+    <ProjectReference Include="..\..\Microsoft.Azure.KeyVault.WebKey\src\Microsoft.Azure.KeyVault.WebKey.csproj" />
```
Comment (weshaggard) [inline sdk/keyvault/Microsoft.Azure.KeyVault.WebKey/tests/Microsoft.Azure.KeyVault.WebKey.Tests.csproj]: Should this just be `..\src\Microsoft.Azure.KeyVault.WebKey.csproj`

File: Directory.Build.props (hunk 1 above)
Comment (weshaggard) [reply Directory.Build.props]: That regex will not work once you move the management libraries under sdk as well. I suggest simply changing the conditions to look like once we need to differentiate the management projects as well. 

```
<IsMgmtProject Condition="'$(IsMgmtProject)' == '' and '$(MSBuildProjectName.Contains('.Management.'))>true<IsMgmtProject>
<IsDataPlaneProject Condition="'$(IsDataPlanProject)' == '' and $(MSBuildProjectName.StartsWith('Microsoft.Azure.')) and '$(IsMgmtProject)' != 'true'>true</IsDataPlaneProject>
```

Comment (chidozieononiwu) [reply Directory.Build.props]: Do I also need to include the new directory in data-plane.proj

Comment (weshaggard) [reply Directory.Build.props]: No I already included it in there when I moved the template so it should work.

File: sdk/AzSdk.reference.props (hunk 2 above)
Comment (chidozieononiwu) [reply sdk/AzSdk.reference.props]: The ones under eng are `reference.targets`. Either way the build broke without bthem so I copied them from the old src path

File: sdk/keyvault/Microsoft.Azure.KeyVault.WebKey/tests/Microsoft.Azure.KeyVault.WebKey.Tests.csproj (hunk 4 above)
Comment (chidozieononiwu) [reply sdk/keyvault/Microsoft.Azure.KeyVault.WebKey/tests/Microsoft.Azure.KeyVault.WebKey.Tests.csproj]: Right, Thanks.

File: sdk/keyvault/Microsoft.Azure.KeyVault.Extensions/tests/Microsoft.Azure.KeyVault.Extensions.Tests.csproj (hunk 3 above)
Comment (chidozieononiwu) [reply sdk/keyvault/Microsoft.Azure.KeyVault.Extensions/tests/Microsoft.Azure.KeyVault.Extensions.Tests.csproj]: Done

**Review by weshaggard — APPROVED** [review]

**Comment by weshaggard** [comment]
FYI @schaabs

File: .azure-pipelines/client.yml [hunk 5]
```diff
@@ -108,7 +108,7 @@ jobs:
 
       - task: DotNetCoreCLI@2
         displayName: 'Build & Test (no live tests)'
-        condition: eq(variables['System.TeamProject'], 'public')
+        condition: eq(variables['System.TeamProject'], 'playground')
```
Comment (weshaggard) [inline .azure-pipelines/client.yml]: I assume this is temporary?

File: eng/AzSdk.test.reference.targets [hunk 6]
```diff
@@ -1,4 +1,5 @@
 <Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="$([MSBuild]::GetPathOfFileAbove('test.props'))" />
```
Comment (weshaggard) [inline eng/AzSdk.test.reference.targets]: Why is that needed we shouldn't be importing that test.props into our projects.

File: .azure-pipelines/client.yml (hunk 5 above)
Comment (chidozieononiwu) [reply .azure-pipelines/client.yml]: Yes, I am just testing it on the playground

File: eng/AzSdk.test.reference.targets (hunk 6 above)
Comment (chidozieononiwu) [reply eng/AzSdk.test.reference.targets]: Was trying to figure out why some tests were failing for no windows OS.

**Comment by schaabs** [comment]
> # Microsoft Azure Keyvault SDK for .NET

As we discussed offline this should be moved to the Microsoft.Azure.KeyVault folder

---
Refers to: sdk/keyvault/README.md:1 in dafd1f2. [](commit_id = dafd1f2c1e72c9f455e08480c681f2e8d1ffa53a, deletion_comment = False)

**Comment by chidozieononiwu** [comment]
> Microsoft Azure Keyvault SDK for .NET
> 
> As we discussed offline this readme should be moved to the Microsoft.Azure.KeyVault folder
> 
> Refers to: sdk/keyvault/README.md:1 in dafd1f2. [](commit_id = dafd1f2, deletion_comment = False)

Done

File: eng/data-plane.proj [hunk 7]
```diff
@@ -2,8 +2,11 @@
   <ItemGroup>
     <!-- Issue https://github.com/Azure/azure-sdk-for-net/issues/5089 some KeyVault tests don't work on linux -->
     <ExcludeDataPlaneProjects Include="$(RepoSrcPath)\KeyVault\data-plane\**\Microsoft.Azure.KeyVault.Tests.csproj" Condition="'$(OS)' != 'Windows_NT'" />
```
Comment (weshaggard) [inline eng/data-plane.proj]: Since this project has been removed you can delete this line now.

Comment (chidozieononiwu) [reply eng/data-plane.proj]: Ok.

**Comment by chidozieononiwu** [comment]
@Azure/azure-sdk-eng

**Comment by chidozieononiwu** [comment]
@weshaggard For some reason I am not able to get the Jenkins build to pass after updating the repo structure.

**Comment by schaabs** [comment]
@chidozieononiwu @weshaggard is this PR ready to merge?  I have some track 2 key vault updates that are waiting on this.

**Comment by chidozieononiwu** [comment]
> @chidozieononiwu @weshaggard is this PR ready to merge? I have some track 2 key vault updates that are waiting on this.

Merged
